src/mscorlib/shared/System/Memory.cs

```diff
@@ -270,9 +270,49 @@ public Span<T> Span
         public bool TryCopyTo(Memory<T> destination) => Span.TryCopyTo(destination.Span);
 
         /// <summary>
-        /// Returns a handle for the array.
-        /// <param name="pin">If pin is true, the GC will not move the array and hence its address can be taken</param>
+        /// Creates a handle for the memory.
+        /// The GC will not move the array until the returned <see cref="MemoryHandle"/>
```

Comment:
The doc comment should not talk about arrays. In the OwnedMemory more there might not be an array.

Reply:
Agreed. The docs should also mention the possibility of GCHandle.Alloc throwing (An instance with nonprimitive (non-blittable) members cannot be pinned.).

I will address this separately after all the changes are in.

```C#
ReadOnlyMemory<object> memory = new object[10];
memory.Pin(); // throws ArgumentException
``